### DISCUSSION TITLE
Is there a way to find out how many threads are active in the thread pool?

### DISCUSSION THREAD

**Discussion by Phoebus888** [redis/lettuce#2784] [Q&A]
My code is as follows
````
    @Override
    public void bindTo(MeterRegistry registry) {
        EventLoopGroupProvider provider = clientResources.eventLoopGroupProvider();
        //Total thread pool
        registry.gauge("lettuce.pool.size", provider, EventLoopGroupProvider::threadPoolSize);
        //Active thread pool
       How does the code implement
    }
````
Now I want to add a method to get the size of the running thread pool to the metrics for monitoring, how to implement

**Comment by mp911de** [comment]
You can implement `EventLoopGroupProvider` to decorate `DefaultEventLoopGroupProvider`.
However, the concept of `EventLoopGroup` is to configure it with a fixed number of threads and the executor service will spin up that many threads so your gauge could be also configured with  the values of `computationThreadPoolSize` and `ioThreadPoolSize` provided to `ClientResources.Builder`.
